src/lib/auth/oauth-client.ts

```diff
@@ -0,0 +1,156 @@
+import http, { IncomingMessage, ServerResponse } from "http";
+const { randomBytes, createHash } = require("node:crypto");
+import url from "url";
+import { AddressInfo } from "net";
+
+export const ACCOUNT_URL =
+  process.env.FAUNA_ACCOUNT_URL ?? "https://account.fauna.com/api/v1";
+
+// Default to prod client id and secret
+const clientId = process.env.FAUNA_CLIENT_ID ?? "Aq4_G0mOtm_F1fK3PuzE0k-i9F0";
+// Native public clients are not confidential. The client secret is not used beyond
+//   client identification. https://datatracker.ietf.org/doc/html/rfc8252#section-8.5
+const clientSecret =
+  process.env.FAUNA_CLIENT_SECRET ??
+  "2W9eZYlyN5XwnpvaP3AwOfclrtAjTXncH6k-bdFq1ZV0hZMFPzRIfg";
```

Comment:
is this really not confidential? what are the defaults used for / do they work for accounts besides yours?

if they are confidential, we should rewrite history to remove references to these (and probably delete & recreate the relevant secret, too).

Reply:
yeah we leave the secret in here to adhere to the api input shape. But it doesn't really serve a purpose
https://datatracker.ietf.org/doc/html/rfc8252#section-8.5

We do what else we can. Enforcing redirect URI's match (localhost)

Similar to github CLI: https://github.com/cli/cli/blob/4f0264e37b2b42253ea866f07b010e5b840aa416/internal/authflow/flow.go#L21

Nothing is stopping someone from distributing their own app with our client ID to try and impersonate Fauna CLI. 

Using PKCE, state, and checking authorization code for a matching client ID help protect from MITM, but that's true for public and confidential clients.